File: design-patterns/go/chain_of_responsibility/cache/cdn_cache_handler.go
// cdn_cache_handler.go

package main

import "fmt"

type CdnCacheHandler struct {
	nextCacheHandler CacheHandler
}

func NewCdnCacheHandler(nextCacheHandler CacheHandler) (cdnCacheHandler *CdnCacheHandler) {
	cdnCacheHandler = &CdnCacheHandler{}
	cdnCacheHandler.nextCacheHandler = nextCacheHandler
	return
}

func (cdnCacheHandler *CdnCacheHandler) HandleRequest(data Data) {
	if data.GetDataType() == CSS || data.GetDataType() == JAVASCRIPT {
		fmt.Printf("Caching file '%v' in CDN\n", data.GetKey())
	} else if cdnCacheHandler.nextCacheHandler != nil {
		cdnCacheHandler.nextCacheHandler.HandleRequest(data)
	}
}

File: design-patterns/php/chainofresponsibility/cache/CdnCacheHandler.php
<?php
// CdnCacheHandler.php

namespace BigBoxCode\DesignPattern\ChainOfResponsibility\Cache;


class CdnCacheHandler extends CacheHandler {
    public function __construct(?CacheHandler $nextCacheHandler) {
        parent::__construct($nextCacheHandler);
    }

    public function handleRequest(Data $data): void {
        if ($data->getType() == DATA_TYPE::CSS || $data->getType() == DATA_TYPE::JAVASCRIPT) {
            echo "Caching file '" . $data->getKey() . "' in CDN\n";
        } else if ($this->nextCacheHandler != null) {
            $this->nextCacheHandler->handleRequest($data);
        }
    }
}